### PR TITLE
[WIP]: (Attempt to) convert all pydicom Dataset variables named `dcm` to `ds` and convert all use of "dcm" in module names, directories, function names etc. to "dicom"

### PR DESCRIPTION
**Pull request by Centrus007**

### CODE REVIEW

**Comment by Centrus007** [comment]
I've also had a go at the (much bigger) task of addressing #56. Passed tests locally and docs looked all in order.

So far I've mostly omitted notebooks, so many will now break. I'll try to tackle them tomorrow. There are also many .json files that will break, but I wasn't really sure where they came from / how important they were (were they from your GUI experimentation a couple of months ago?)

**Comment by Centrus007** [comment]
Since I made a rather large set of changes since your approval, I'll leave this unmerged until you get a chance to see it :)

**Comment by Centrus007** [comment]
> I am partially of the opinion to not worry too much about updating the notebooks. But I am torn, what do you think?

I think it'll be relatively easy to adjust them as we use them...

> One request before merge, we need to mark this under breaking changes within the change log that is in the root directory. This will be a breaking changes that is involved with the upgrade from 0.5.* --> 0.6.0.

> For a different branch we also need to make it so that that change log appears on the documentation website before the 0.6.0 release. Given users would need to update their code.

Good point. I'd forgotten about the Changelog and am not too familiar with adjusting its visibility in the docs. I'll have a crack.

**Comment by SimonBiggs** [comment]
With regards to putting the change log within the docs we have two options.
One would be to include the following sphinx module as a dependency with
building our sphinx docs which allows us to also pull in markdown files:

https://github.com/rtfd/recommonmark

Another option would be to convert the change log to reST and then directly
import it.

Of the two options I am leaning towards the former as I am sure there will
be multiple cases where we will want to use markdown within the docs for
some reason.

**Comment by SimonBiggs** [comment]
As a note, when I was first setting up the docs I initially had the readme
as a markdown file and was importing it into the docs. Here is a commit
where that was the case:

https://github.com/pymedphys/pymedphys/tree/69a2a20f86c7c2c4ccdcbc04f49f16c78e137a58

**Comment by Centrus007** [comment]
Shall I merge these in or sort out changelog visibility first?

File: changelog.md [hunk 1]
```diff
@@ -27,7 +36,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - nil
 
 
-## [0.5.1] - 2018-01-05
+## [0.5.1] - 2019-01-05
```
Comment (Centrus007) [inline changelog.md]: I think this should read 2019. @SimonBiggs, can you confirm?

**Comment by SimonBiggs** [comment]
> Shall I merge these in or sort out changelog visibility first?

Merge this first. Changlog visibility can be a separate branch.

File: changelog.md (hunk 1 above)
Comment (SimonBiggs) [reply changelog.md]: Yup 2019. See https://pypi.org/project/pymedphys/#history

**Comment by SimonBiggs** [comment]
As an aside I think we should draw inspiration from how other projects do their changelogs:

https://github.com/angular/angular/blob/master/CHANGELOG.md#breaking-changes

**Pull request merged**